File: SmartProject/SmartProjectRepo/SmartProject/Models/Task/TaskModel.cs
using SmartProject.Models.Task;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SmartProject.Models
{
    public class TaskModel
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public UserBasicInfo UserCreated { get; set; }
        public UserBasicInfo? UserAssigned { get; set; }
        public decimal? EstimatedTime { get; set; }      
        public DateTime AddedDate { get; set; }
        public DateTime? DeadlineDate { get; set; }
        public DateTime ModifiedDate { get; set; }
        public TaskTypeModel Type { get; set; }
        public TaskStatusModel Status { get; set; }
        public TaskPriorityModel Priority { get; set; }
        public ReleaseModel Release { get; set; }
        public string Description { get; set; }
        public ICollection<TaskCommentModel> TaskComments { get; set; }
    }
}

File: SmartProject/SmartProjectRepo/material-angular-dashboard/src/app/models/task/task.model.ts
import { TaskPriority } from "./task-priority.model";
import { TaskType } from "./task-type.model";
import { TaskStatus } from "./task-status.model";
import { User } from "../user.model";

export class Task {
    id?: number;
    title: string;
    author: User;
    addedDate: string;
    type: TaskType;
    status: TaskStatus;
    priority: TaskPriority;
}